### PR TITLE
op-challenger: Replace duplicate tx sending code with TxSender

### PR DESCRIPTION
**Pull request by ajsutton**
**Description**

Replaces a number of duplicate `sendAndWait` functions with a `TxSender` that provides a convenient wrapper around the TxManager.  This also allows us to apply a global limit on the number of pending transactions.

**Tests**

Added unit tests.

**Metadata**

- https://github.com/ethereum-optimism/client-pod/issues/480 (because otherwise we'd need to add a fourth copy of sendAndWait)

### CODE REVIEW

File: op-challenger/game/fault/preimages/direct.go [hunk 1]
```diff
@@ -38,23 +38,15 @@ func (d *DirectPreimageUploader) UploadPreimage(ctx context.Context, claimIdx ui
...
-		return err
-	}
-	if receipt.Status == ethtypes.ReceiptStatusFailed {
-		d.log.Error("DirectPreimageUploader tx successfully published but reverted", "tx_hash", receipt.TxHash)
-	} else {
-		d.log.Debug("DirectPreimageUploader tx successfully published", "tx_hash", receipt.TxHash)
-	}
-	return nil
+func (d *DirectPreimageUploader) sendTxAndWait(candidate txmgr.TxCandidate) error {
+	_, err := d.txSender.SendAndWait(candidate)
```
Comment (refcell) [inline op-challenger/game/fault/preimages/direct.go]: Should we log the tx receipt here?

File: op-challenger/game/fault/preimages/large.go [hunk 2]
```diff
@@ -137,43 +137,23 @@ func (p *LargePreimageUploader) initLargePreimage(ctx context.Context, uuid *big
...
-		}
-		if receipt.Receipt.Status == ethtypes.ReceiptStatusFailed {
-			p.log.Error("LargePreimageUploader add leafs tx successfully published but reverted", "tx_hash", receipt.Receipt.TxHash)
-		} else {
-			p.log.Debug("LargePreimageUploader add leafs tx successfully published", "tx_hash", receipt.Receipt.TxHash)
-		}
-	}
-	return nil
+	_, err := p.txSender.SendAndWait(txs...)
+	return err
```
Comment (refcell) [inline op-challenger/game/fault/preimages/large.go]: Should we change the `LargePreimageUploader`'s `sendTxAndWait` method to accept variadic tx candidate args and then just pass in there with an error return? The `sendTxAndWait` method could then also log any receipt data if we want.

**Review by refcell — APPROVED** [review]

File: op-challenger/game/fault/preimages/direct.go (hunk 1 above)
Comment (ajsutton) [reply op-challenger/game/fault/preimages/direct.go]: The txSender logs when the tx is successful in the same way the old code did. That's actually the majority of the code we kept duplicating.

File: op-challenger/game/fault/preimages/large.go (hunk 2 above)
Comment (ajsutton) [reply op-challenger/game/fault/preimages/large.go]: Given `TxSender` is handling the logging, I inlined a couple of these `sendTxAndWait` functions which were just stripping the returned receipts.  In `responder` it's still useful to do that as the error is returned directly so it's neater to have a function that only returns an error.  For this and `direct` it's easy to ignore the returned receipts inline so don't need the method.

**Comment by refcell** [comment]
This is very nice

**Pull request merged**